src/data_hub/api.py

```diff
@@ -890,15 +974,25 @@ def get_dataframe(
     # TODO: list_database_rows and describe_row can be called in parallel
 
     # figure out the rows
-    rows = _api.list_database_rows(
-        database_row_id=database_id,
-        client=client,
-    )
+    if filter is None:
+        rows = _api.list_database_rows(
+            database_row_id=database_id,
+            client=client,
+            _stash=_stash,
+        )
+    else:
+        rows = _api.list_database_rows(
+            database_row_id=database_id,
+            client=client,
+            _stash=_stash,
+            filter=filter,
+        )
```

Comment:
Is this really the way to pass an optional param? Can't `list_database_rows` handle an filter value of None?

Reply:
i get a bad request error

```
BadRequestError: Error code: 400 - {'errors': [{'title': 'invalid request', 'status': 400, 'meta': {'issues': [{'code': 'invalid_union', 'unionErrors': [{'issues': [{'code': 'invalid_union', 'unionErrors': [{'issues': [{'code': 'invalid_type', 'expected': 'object', 'received': 'null', 'path': ['filter'], 'message': 'Expected object, received null'}], 'name': 'ZodError'}, {'issues': [{'code': 'invalid_type', 'expected': 'object', 'received': 'null', 'path': ['filter'], 'message': 'Expected object, received null'}], 'name': 'ZodError'}, {'issues': [{'code': 'invalid_type', 'expected': 'object', 'received': 'null', 'path': ['filter'], 'message': 'Expected object, received null'}], 'name': 'ZodError'}, {'issues': [{'code': 'invalid_type', 'expected': 'object', 'received': 'null', 'path': ['filter'], 'message': 'Expected object, received null'}], 'name': 'ZodError'}], 'path': ['filter'], 'message': 'Invalid input'}], 'name': 'ZodError'}, {'issues': [{'code': 'invalid_type', 'expected': 'object', 'received': 'null', 'path': ['filter'], 'message': 'Expected object, received null'}], 'name': 'ZodError'}, {'issues': [{'code': 'invalid_type', 'expected': 'object', 'received': 'null', 'path': ['filter'], 'message': 'Expected object, received null'}], 'name': 'ZodError'}, {'issues': [{'code': 'invalid_type', 'expected': 'object', 'received': 'null', 'path': ['filter'], 'message': 'Expected object, received null'}], 'name': 'ZodError'}], 'path': ['filter'], 'message': 'Invalid input'}]}}]}
```

Reply:
When you pass `None`, instead of converting to `null`, can you not set the key?